libraries/AP_Math/AP_Math.cpp

```diff
@@ -67,14 +67,17 @@ template float safe_asin<short>(const short v);
 template float safe_asin<float>(const float v);
 template float safe_asin<double>(const double v);
 
+// sqrt which takes any type and returns 0 if the input is NaN or less than zero
 template <typename T>
 float safe_sqrt(const T v)
 {
-    float ret = sqrtf(static_cast<float>(v));
-    if (isnan(ret)) {
-        return 0;
+    // cast before checking so we sqrtf the same value we check
+    const float val = static_cast<float>(v);
```

Comment:
Wow, we went through this entire discussion without noting that we're squashing any double which might be passed in....

Reply:
I had noticed that. But we do it in several places, I'm not too concerned specifically here.